### ISSUE TITLE
Verbose type annotations in help panel

### ISSUE THREAD

**Issue by thangleiter** [spyder-ide/spyder#22249]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
The help panel in my opinion is one of the nicest features in Spyder. With the introduction and proliferation of type annotations in Python, it became even more useful if developers did not document expected parameter types in their docstrings. More recently though, with Python making the typing system more complex as well as NumPy introducing typing to their codebase, function signatures become more and more bloated when annotated with complex union types. Note that this is of course a quite general issue that does not affect Spyder alone; the IPython tooltip help for example behaves the same way.

An example: 

```python
from collections.abc import Sequence
type_alias = Sequence[int | str] | int | str

def my_fun(x: type_alias) -> type_alias:
    do_whatever_with(x)
```
<details><summary>Rendered help</summary>

![grafik](https://github.com/spyder-ide/spyder/assets/32895729/d8344930-b2ee-4d87-ba18-a8cd4bd5548b)

</details>

Now, one can argue if type aliases defined in such a way are the way to go. The most relevant situation in my case is when using `numpy.typing` to annotate functions:

```python
import numpy.typing as npt

def my_np_fun(x: npt.ArrayLike, axis: int) -> npt.NDArray[float]:
    return np.asarray(x, np.float64).sum(axis)
```
<details><summary>Rendered help</summary>

![grafik](https://github.com/spyder-ide/spyder/assets/32895729/151e9399-4f72-4900-929a-3171c2a4a4b0)

</details>

In this case, it becomes really quite hard to figure out even that the `axis` argument exists because it is hard to visually distinguish it from the extremely long type annotation of the previous argument. Again, it may be argued that this is an upstream issue, either with Sphinx and the way it handles verbose annotations or with the way the type aliases are implented. Still, the current situation makes the help explorer much less useful to me.

As far as I can tell, the only way to prevent this level of verbosity in Sphinx is to manually specify type alias substitutions for the `autodoc` extension* in [`conf.py`](https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/help/utils/conf.py) as documented [here](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases). This is not very user-friendly, but at least it could cover the most extreme cases like `npt.ArrayLike`.

*If it is even used? How else does Spyder/sphinx render type annotations?

That would look something like this:
```python
autodoc_type_aliases = {
    'ArrayLike': 'numpy.typing.ArrayLike',
    'DTypeLike': 'numpy.typing.DTypeLike',
    'NBitBase': 'numpy.typing.NBitBase',
    'NDArray': 'numpy.typing.NDArray',
}
```
Since users might not use relative imports, one might also want to add the common alias:
```python
autodoc_type_aliases |= {f"npt.{key}": value for key, value in autodoc_type_aliases.items()}
```

One might also imagine this to be configured through a settings page where users can add custom aliases they don't want explicitly resolved.

**Comment by dalthviz** [comment]
Hi @thangleiter thank you for the feedback and thank you for describing your use case and possible ideas to improve the Help pane functionality in such a clear way! I think that exploring the idea of adding a setting that enables you to exclude type aliases from being rendered could be worthy :+1:

From the sphinx extension part, I would guess `autodoc` is used indirectly via [`autosummary`](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#module-sphinx.ext.autosummary) ? 🤔 

What do you think @spyder-ide/core-developers ?

**Comment by ccordoba12** [comment]
I agree with @thangleiter: we definitely need to improve our support for typing in the Help pane.

We'll try to take a look at it after we release Spyder 6.0 in a few weeks.

**Comment by thangleiter** [comment]
> Hi @thangleiter thank you for the feedback and thank you for describing your use case and possible ideas to improve the Help pane functionality in such a clear way! I think that exploring the idea of adding a setting that enables you to exclude type aliases from being rendered could be worthy 👍
> 
> From the sphinx extension part, I would guess `autodoc` is used indirectly via [`autosummary`](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#module-sphinx.ext.autosummary) ? 🤔

I believe so, yes.

On a related note: If going down the route of adding a sphinx configuration page in settings, I think the `autodoc_typehints` option would also be worthwile. This configures if type hints are displayed only in the function `'signature'`, only in the `'description'`, as well as `'both'` or `'none'`.
